optaplanner-wb-screens/optaplanner-wb-domain-editor/optaplanner-wb-domain-editor-backend/src/main/java/org/optaplanner/workbench/screens/domaineditor/backend/server/PlannerMethodFilter.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.workbench.screens.domaineditor.backend.server;
+
+import javax.annotation.Generated;
+import javax.enterprise.context.ApplicationScoped;
+
+import org.jboss.forge.roaster.model.Method;
+import org.kie.workbench.common.services.datamodeller.driver.MethodFilter;
+
+@ApplicationScoped
+public class PlannerMethodFilter implements MethodFilter {
+
+    @Override
+    public boolean accept( Method<?, ?> method ) {
+        return !method.isConstructor() && method.getAnnotation( Generated.class ) != null;
```

Comment:
Hello @mcimbora , I think it can be worth to provide a more fine grained filtering here. To be sure that the method  is not a constructor, has the @Generated annotation, and also "is one of the generated" by the planner stuff. Probably inspecting some additional planner generated annotation for the method (I I don't remember wrong for this planner generated method you added has also a planner annotation with two parameters?) or eventually exploring the return type, or some other trick. This will prevent us from loading other methods with the @Generated annotation that we don't know how to manage. Specially because when the java source is being updated, all the accepted methods are being deleted.